app/ui/browser-australis/views/browser/decorations/statusbar.jsx

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import { connect } from 'react-redux';
+
+import Style from '../../../../shared/style';
+
+import * as UIConstants from '../../../constants/ui';
+import * as UISelectors from '../../../selectors/ui';
+
+const STATUS_STYLE = Style.registerStyle({
+  position: 'fixed',
+  bottom: 0,
+  left: 0,
+  zIndex: UIConstants.STATUS_BAR_ZINDEX,
+  padding: '2px 8px',
+  backgroundColor: 'var(--theme-window-background)',
+  color: 'var(--theme-content-color)',
+});
+
+class StatusBar extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
```

Comment:
On Tue, Sep 6, 2016 at 1:30 PM Victor Porof <notifications@github.com>
wrote:

> In app/ui/browser-australis/views/browser/decorations/statusbar.jsx
> <https://github.com/mozilla/tofino/pull/1054#discussion_r77624509>:
>
> > +import * as UISelectors from '../../../selectors/ui';
> > +
> > +const STATUS_STYLE = Style.registerStyle({
> > +  position: 'fixed',
> > +  bottom: 0,
> > +  left: 0,
> > +  zIndex: UIConstants.STATUS_BAR_ZINDEX,
> > +  padding: '2px 8px',
> > +  backgroundColor: 'var(--theme-window-background)',
> > +  color: 'var(--theme-content-color)',
> > +});
> > +
> > +class StatusBar extends Component {
> > +  constructor(props) {
> > +    super(props);
> > +    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
>
> Based on that PR, I understand that simply extending PureComponent will
> also cause all children to use shallow compare. Not exactly the same
> semantics as declaring our own shouldComponentUpdate.
>

But the same as `this.shouldComponentUpdate =
PureRenderMixin.shouldComponentUpdate.bind(this);` though?

Joe.